### PR TITLE
GAWB-3272: entity statistics in admin-stats endpoint

### PR DESCRIPTION
**Pull request by davidangb**
add entity counts to the admin-statistics endpoint. This is the first step towards GAWB-3272. I plan to consume these counts from orchestration (which will calculate other stats as well) and push from there to kibana.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala [hunk 1]
```diff
@@ -743,5 +743,22 @@ trait EntityComponent {
...
+
+    object EntityStatisticsQueries extends RawSqlQuery {
+      val driver: JdbcDriver = EntityComponent.this.driver
+      def countEntitiesofTypeInNamespace(workspaceNamespace: Option[String], workspaceName: Option[String]): ReadAction[Seq[NamedStatistic]] = {
+        val baseSelect = sql"""select entity_type, count(1) as entityCount
+                from ENTITY e, WORKSPACE w
+                where e.workspace_id = w.id
+                  and e.deleted = 0"""
+        val namespaceClause = workspaceNamespace.map(ns => sql" and w.namespace = $ns").getOrElse(sql"")
+        val nameClause = workspaceName.map(n => sql" and w.name = $n").getOrElse(sql"")
```
Comment (rushtong) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala:790]: I can't tell if `workspaceNamespace` and `workspaceName` are protected against sql injection here. It doesn't look like `concatSqlActions` does anything to the string values in the name clauses. Is this handled by the `sql` macro?

**Review by rushtong — APPROVED** [review]
👍 Was not able to run rawls locally, but the code looks good aside from one question about how we're protecting against sql-injection.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala (hunk 1 above)
Comment (MatthewBemis) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala:790]: yeah, the `sql` macro turns all variables into bind variables to protect against this

see http://slick.lightbend.com/doc/3.0.0/sql.html#string-interpolation

Comment (rushtong) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala:790]: 👍  Gotcha, thanks!

**Review by MatthewBemis — APPROVED** [review]

**Review by rushtong — APPROVED** [review]
Still 👍 Needs a rebase to run locally.

**Comment by davidangb** [comment]
looks like a potentially-intermittent rawls build failure, so jenkins retest.

**Comment by davidangb** [comment]
rawls-build failure:
```
[info] *** 1 TEST FAILED ***
[error] Failed: Total 870, Failed 1, Errors 0, Passed 869, Ignored 2
[error] Failed tests:
[error] 	org.broadinstitute.dsde.rawls.monitor.BucketDeletionMonitorSpec

[info] BucketDeletionMonitor
[info] - should delete buckets *** FAILED ***
[info]   The code passed to eventually never returned normally. Attempted 51 times over 1.014300999 seconds. Last failure message: 
[info]   googleServicesDAO.deleteBucket(
[info]       "empty-bucket"
[info]   );
[info]   Wanted 1 time:
[info]   -> at org.broadinstitute.dsde.rawls.monitor.BucketDeletionMonitorSpec.$anonfun$new$2(BucketDeletionMonitorSpec.scala:50)
[info]   But was 2 times. Undesired invocation:
[info]   -> at org.broadinstitute.dsde.rawls.monitor.BucketDeletionMonitor.deleteBucket(BucketDeletionMonitor.scala:52)
[info]   . (BucketDeletionMonitorSpec.scala:49)
```

**Pull request merged**